sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -192,43 +198,51 @@ public static void login(@NonNull String jwt, KinCallback<Void> loginCallback) {
 	private static void internalLogin(@NonNull final String jwt, final KinCallback<Void> loginCallback) {
 		try {
 			checkInstanceNotNull();
-			if(!AuthRepository.getInstance().isSameUser(jwt)) {
-				logout();
+			@UserLoginState int loginState = AuthRepository.getInstance().getUserLoginState(jwt);
+			switch (loginState) {
+				case UserLoginState.DIFFERENT_USER:
+					logout();
+				case UserLoginState.FIRST:
+					isAccountLoggedIn.getAndSet(false);
+					eventLogger.send(UserLoginRequested.create());
+					break;
+				case UserLoginState.SAME_USER:
+					break;
 			}
 			AuthRepository.getInstance().setJWT(jwt);
 		} catch (final ClientException exception) {
 			sendLoginFailed(exception, loginCallback);
 		}
 
-		AuthRepository.getInstance().getAuthToken(new KinCallback<AuthToken>() {
+		AuthRepository.getInstance().getAccountInfo(new KinCallback<AccountInfo>() {
 			@Override
-			public void onResponse(AuthToken authToken) {
+			public void onResponse(AccountInfo accountInfo) {
 				String publicAddress;
 				try {
-					BlockchainSourceImpl.getInstance().loadAccount(authToken.getEcosystemUserID());
+					BlockchainSourceImpl.getInstance().loadAccount(accountInfo.getAuthToken().getEcosystemUserID());
 					publicAddress = BlockchainSourceImpl.getInstance().getPublicAddress();
 				} catch (final BlockchainException exception) {
 					sendLoginFailed(exception, loginCallback);
 					return;
 				}
 				AccountManagerImpl.getInstance().start();
```

Comment:
I think that there is no need to call `AccountManagerImpl.getInstance().start();` incase of failure (it will listen to blockchain for no reason) so it better to place it in sendLoginSucceed